src/Transient/Internals.hs

```diff
@@ -265,328 +261,331 @@ readWithErr line=
+  r    <- runTrans procedure
+  modify $ \s -> s { maxThread = msem } -- restore it
+  return r
+
+-- | Delete all the previous children generated by the expression taken as
+-- parameter and continue execution of the current thread.
+oneThread :: TransIO a -> TransIO a
+oneThread comp = do
+  chs <- liftIO $ newTVarIO []
+  r   <- comp
+  modify $ \ s -> s { children = chs }
+  killChilds
+  return r
 
+-- | Display the active threads in a tree-like structure, highlighting the
+-- parent-child relationships between the threads.
 showThreads :: TransIO empty
-showThreads= do
-   st' <- gets (fromJust . parent)
-   liftIO $ showTree 0 st'
-   stop
-   where
-   toplevel st =
-      case parent st of
-        Nothing ->  st
-        Just p -> toplevel p
-
-   showThreads' n rchs= do
-      chs <- atomically $ readTVar rchs
-      mapM_ (showTree n) chs
-
-   showTree n ch=  do
-         putStr $ take n $ repeat  ' '
-         print $ threadId ch
-         showThreads' (n+4) $ children ch
-
-
--- | add n threads to the limit of threads. If there is no limit, it set it
+showThreads = do
+  st' <- gets (fromJust . parent)
+  liftIO $ showTree 0 st'
+  stop
+  where toplevel st =
+          case parent st of
+            Nothing -> st
+            Just p  -> toplevel p
+
+        showThreads' n rchs = do
+          chs <- atomically $ readTVar rchs
+          mapM_ (showTree n) chs
+
+        showTree n ch = do
+          putStr $ take n $ repeat ' '
+          print $ threadId ch
+          showThreads' (n + 4) $ children ch
+
+-- | Add n threads to the limit of threads. If there is no limit, set it.
 addThreads' :: Int -> TransIO ()
-addThreads' n= Transient $ do
-   msem <- gets maxThread
-   case msem of
+addThreads' n = Transient $ do
+  msem <- gets maxThread
+  case msem of
     Just sem -> liftIO $ modifyIORef sem $ \n' -> n + n'
     Nothing  -> do
-        sem <- liftIO (newIORef n)
-        modify $ \ s -> s{maxThread= Just sem}
-   return $ Just ()
-
--- | assure that at least there are n threads available
-addThreads n= Transient $ do
-   msem <- gets maxThread
-   case msem of
-     Nothing -> return ()
-     Just sem ->  liftIO $ modifyIORef sem $ \n' -> if n' > n then n' else  n
-   return $ Just ()
+      sem <- liftIO (newIORef n)
+      modify $ \ s -> s { maxThread = Just sem }
+  return $ Just ()
+
+-- | Ensure that at least n threads are available.
+ensureThreads :: Int -> TransIO ()
+ensureThreads n = Transient $ do
+  msem <- gets maxThread
+  case msem of
+    Nothing  -> return ()
+    Just sem -> liftIO $ modifyIORef sem $ \n' -> if n' > n then n' else n
+  return $ Just ()
+
+{-# DEPRECATED addThreads "Use `ensureThreads` instead." #-}
+addThreads :: Int -> TransIO ()
+addThreads = ensureThreads
+
 --getNonUsedThreads :: TransIO (Maybe Int)
 --getNonUsedThreads= Transient $ do
 --   msem <- gets maxThread
 --   case msem of
 --    Just sem -> liftIO $ Just <$> readIORef sem
 --    Nothing -> return Nothing
 
```

Comment:
It's always safe to add a fallback for old codebases about API naming changes so that's why I kept the old 'addThreads'. I'm concerned that `transient-universe`, a heavy user of `transient`, might use most of the functions in this module. What we can do is go with the next release with the deprecated message and in a future release completely remove the old function as a breaking change.